### PR TITLE
fix: use simprocs in step-theorem generation

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

Some stepi-theorems were being generated which still included elements like `decode_bit_masks 0#1 3#6 0#6 false`.
After investigation, I noticed simprocs were *not* being passed to `simp` in `StepThms`, so I fixed that.

### Testing:

`make all` ran locally

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Proofs/Experiments/AbsVCG.lean [hunk 1]
```diff
@@ -274,6 +272,7 @@ theorem program_effects_lemma (h_pre : abs_pre s0)
             BitVec.truncate 64 1#32)) := by
     simp (config := {decide := true}) only [h_step_4, h_step_3, h_step_2, h_step_1,
                                             state_simp_rules]
+    rfl
```
Comment (alexkeizer) [inline Proofs/Experiments/AbsVCG.lean:275]: Not sure why this rfl is now needed, but the proof broke without it

**Comment by bollu** [comment]
LGTM

File: Proofs/SHA512/Sha512DecodeExecLemmas.lean [hunk 2]
```diff
@@ -25,8 +25,7 @@ info: sha512_program.exec_0x126c90 (s : ArmState) :
       (ArmInst.BR (BranchInst.Compare_branch { sf := 1#1, _fixed := 26#5, op := 1#1, imm19 := 523804#19, Rt := 2#5 }))
       s =
     w StateField.PC
-      (if decide (r (StateField.GPR 2#5) s = BitVec.zero (if 1#1 = 1#1 then 64 else 32)) = false then
-        r StateField.PC s + BitVec.signExtend 64 (523804#19 <<< 2)
+      (if decide (r (StateField.GPR 2#5) s = BitVec.zero 64) = false then r StateField.PC s + 18446744073709549680#64
```
Comment (shigoel) [inline Proofs/SHA512/Sha512DecodeExecLemmas.lean:28]: Nice! I've been so tired of seeing ground terms like `(if 1#1 = 1#1 then 64 else 32)`. Guess the wrong way of passing in `simprocs` was the culprit all along.

File: Proofs/Experiments/AbsVCG.lean (hunk 1 above)
Comment (shigoel) [reply Proofs/Experiments/AbsVCG.lean:275]: It's brittle anyway -- I'll redo all of it later. Thanks!

**Review by shigoel — APPROVED** [review]
LGTM, thanks, @alexkeizer!

**Pull request merged**